base/sparse/umfpack.jl

```diff
@@ -113,12 +148,13 @@ function lufact{Tv<:UMFVTypes,Ti<:UMFITypes}(S::SparseMatrixCSC{Tv,Ti})
     finalizer(res, umfpack_free_symbolic)
     umfpack_numeric!(res)
 end
-lufact(A::SparseMatrixCSC) = lufact(float(A))
+lufact{T<:Real}(A::SparseMatrixCSC{T}) = lufact(map(Float64, A))
```

Comment:
Use `convert` here
```jl
julia> A = sparse(rand(1:1000, 2000), rand(1:500, 2000), rand(-10:10, 2000));

julia> @time map(Float64, A);
  0.042705 seconds (500.01 k allocations: 7.663 MB, 9.65% gc time)

julia> @time convert(SparseMatrixCSC{Float64}, A);
  0.000035 seconds (25 allocations: 16.281 KB)
```
We might also want to throw on big types because we usually don't want to reduce precision silently.